lib/discovery/index.js

```diff
@@ -832,7 +831,7 @@ export class PeerInfo extends TypedEmitter {
    * @typedef {Object} PeerInfoOptions
    * @property {IdentityPublicKeyString} identityPublicKey
    * @property {('dht'|'mdns')} discoveryType
-   * @property {ConnectionStream} connection
+   * @property {SecretStream<RawConnectionStream>} connection
```

Comment:
ideally would do some conditional typing to narrow down the RawConnectionStream type based on the `discoveryType`. not sure how to go about that in TS + JSDoc